cmd/cliVolumeOps.go

```diff
@@ -33,28 +33,47 @@ type cliVolumeInputs struct {
 	// if namespace is "", use all-namespaces
 	// else use specified namespace
 	namespace *string
+	labels    map[string]string
 }
 
 type cliVolumeOps struct {
 	cliVolumeInputs
 	pxVolumeOps portworx.PxVolumeOps
 }
 
-// Look for all of the common flags and create a new cliVolumeInputs object
+// GetCliVolumeInputs looks for all of the common flags and create a new cliVolumeInputs object
 func GetCliVolumeInputs(cmd *cobra.Command, args []string) *cliVolumeInputs {
 	showK8s, _ := cmd.Flags().GetBool("show-k8s-info")
 	output, _ := cmd.Flags().GetString("output")
 	showLabels, _ := cmd.Flags().GetBool("show-labels")
 	namespace := string("")
-	return &cliVolumeInputs{
-		BaseFormatOutput: util.BaseFormatOutput{
-			FormatType: output,
-		},
-		showK8s:     showK8s,
-		showLabels:  showLabels,
-		volumeNames: args,
-		namespace:   &namespace, // In most places use all namespaces
+	labels, _ := cmd.Flags().GetString("selector")
+	//convert string to map
+	mlabels, _ := util.CommaStringToStringMap(labels)
+	// If valid label is present, we need to pass it.
+	if len(mlabels) != 0 {
```

Comment:
I don't think you need the if statement. You can just return the object of line 55 because either way it will be nil, zero len, or have something, and it can be checked by the user of this info.

Reply:
Wanted to be doubly sure so had this check. Will remove it.